### PR TITLE
Maven project and Java source for Ecosystem builder

### PR DESCRIPTION
**Pull request by Roavill2**
After uploading jar file and resources on it's own, I have now made sure to upload the source, inside a maven project.

### CODE REVIEW

File: galasa-extensions-parent/dev.galasa.ecosystem.builder/src/main/java/galasa/ecosystem/docker/builder/DockerEcosystemBuilder.java [hunk 1]
```diff
@@ -0,0 +1,628 @@
...
+		  while (true) {
+			    apiPort = scanner.nextLine();
+			    if (apiPort.isEmpty() || apiPort.equals("8181:8181")) {
+			        apiPort = "8181:8181";
+			        break;
+			    } else {
+			        try {
+			        	String port1 = null;
+			        	String port2 = null;
+			        	port1 = apiPort.substring(0, apiPort.indexOf(":"));
```
Comment (Michael-Baylis) [inline galasa-extensions-parent/dev.galasa.ecosystem.builder/src/main/java/galasa/ecosystem/docker/builder/DockerEcosystemBuilder.java]: not that you need this code anymore,  but should have been a regex check

File: galasa-extensions-parent/dev.galasa.ecosystem.builder/src/main/resources/docker/bootstrap.properties [hunk 2]
```diff
@@ -0,0 +1,2 @@
+framework.config.store=etcd:http://172.21.0.1:2379
```
Comment (Michael-Baylis) [inline galasa-extensions-parent/dev.galasa.ecosystem.builder/src/main/resources/docker/bootstrap.properties]: Transient files should not be committed

File: galasa-extensions-parent/dev.galasa.ecosystem.builder/src/main/java/galasa/ecosystem/docker/builder/DockerEcosystemBuilder.java [hunk 3]
```diff
@@ -0,0 +1,628 @@
...
+			        } catch (NumberFormatException | StringIndexOutOfBoundsException ex) {
+			            System.out.println("Please enter a port in the format nnnn:nnnn");
+			        }
+			    }
+			}
+		  
+		  System.out.println("Setting port to: " + sbPort + "\n");
+		  
+		  System.out.println("Docker config set up complete..." + "\n");
+
```
Comment (Michael-Baylis) [inline galasa-extensions-parent/dev.galasa.ecosystem.builder/src/main/java/galasa/ecosystem/docker/builder/DockerEcosystemBuilder.java]: At this point you should present an option of all the settings and ask them if it is ok
if not, give them the option to change which bit and come back here

File: galasa-extensions-parent/dev.galasa.ecosystem.builder/src/main/java/galasa/ecosystem/docker/builder/DockerEcosystemBuilder.java [hunk 4]
```diff
@@ -0,0 +1,628 @@
...
+		  
+		  System.out.println("Setting port to: " + sbPort + "\n");
+		  
+		  System.out.println("Docker config set up complete..." + "\n");
+		  
+		  System.out.println("Updating Ecosystem config file..." + "\n");
+		  
+		  ProcessBuilder processBuilder = new ProcessBuilder();
+
+	        processBuilder.command("./scripts/docker-update.sh", subnet, gateway, apiName, netName, apiResPol, apiVol, apiVolSrc, apiVolTrg, apiPort, contName, contResPol, contVol, cpsName, cpsResPol, cpsVolSrc, cpsVolTrg, cpsPort, cdbiName, cdbiVolSrc, cdbiVolTrg, cdbName, cdbResPol, cdbEnv, cdbVolSrc, cdbVolTrg, cdbPort, resmonName, resmonResPol, resName, resResPol, resPort, sbName, sbResPol, sbPort);
```
Comment (Michael-Baylis) [inline galasa-extensions-parent/dev.galasa.ecosystem.builder/src/main/java/galasa/ecosystem/docker/builder/DockerEcosystemBuilder.java]: Dont run the script,  they need to do that.   if you are going to run it,  then you need to analyse the output and report what is wrong.

File: galasa-extensions-parent/dev.galasa.ecosystem.builder/src/main/java/galasa/ecosystem/docker/builder/DockerEcosystemBuilder.java [hunk 5]
```diff
@@ -0,0 +1,628 @@
...
+		Scanner scanner = new Scanner(System.in);
+		System.out.println("Welcome to the Galasa Ecosystem for Docker set up" + "\n");
+		System.out.println("Would you like to reuse old config (default Y):");
+		String reuse = scanner.nextLine();
+	    while (!(reuse.toUpperCase().equals("Y") || reuse.toUpperCase().equals("N") || reuse.isEmpty())) {	
+			System.out.println("Please enter Y or N");
+			reuse = scanner.nextLine();
+		}
+		if ((reuse.toUpperCase().equals("Y") || (reuse.isEmpty())))	{
+			System.out.println("Using previously set config");
```
Comment (Michael-Baylis) [inline galasa-extensions-parent/dev.galasa.ecosystem.builder/src/main/java/galasa/ecosystem/docker/builder/DockerEcosystemBuilder.java]: This should present a list of options, their current values and give them the opportunity to change individual options.
At the moment,  if they make a mistake in a reply,  they have to ctrl+c out and start all over,  not very user friendly.

File: galasa-extensions-parent/dev.galasa.ecosystem.builder/src/main/java/galasa/ecosystem/docker/builder/DockerEcosystemBuilder.java [hunk 6]
```diff
@@ -0,0 +1,628 @@
...
+			
+			writer.close();
+		  
+        }
+		
+		scanner.close();
+		
+		System.out.println("Executing ecosystem set up..." + "\n");
```
Comment (Michael-Baylis) [inline galasa-extensions-parent/dev.galasa.ecosystem.builder/src/main/java/galasa/ecosystem/docker/builder/DockerEcosystemBuilder.java]: Should have saved the responses in a YAML file so can be reused

File: galasa-extensions-parent/dev.galasa.ecosystem.builder/src/main/resources/docker/config.properties [hunk 7]
```diff
@@ -0,0 +1,67 @@
+subnet=172.21.0.0/16
```
Comment (Michael-Baylis) [inline galasa-extensions-parent/dev.galasa.ecosystem.builder/src/main/resources/docker/config.properties]: Transient files should not be committed.
If this is a default responses files,  then should be yaml

File: galasa-extensions-parent/dev.galasa.ecosystem.builder/src/main/resources/docker/controller.properties [hunk 8]
```diff
@@ -0,0 +1,4 @@
+engine_image=docker.galasa.dev/galasa-boot-embedded-amd64:0.5.0
```
Comment (Michael-Baylis) [inline galasa-extensions-parent/dev.galasa.ecosystem.builder/src/main/resources/docker/controller.properties]: Transient files should not be committed

File: galasa-extensions-parent/dev.galasa.ecosystem.builder/src/main/resources/docker/galasa-env.properties [hunk 9]
```diff
@@ -0,0 +1,2 @@
+COUCHDB_USER=galasa
```
Comment (Michael-Baylis) [inline galasa-extensions-parent/dev.galasa.ecosystem.builder/src/main/resources/docker/galasa-env.properties]: Transient files should not be committed

File: galasa-extensions-parent/dev.galasa.ecosystem.builder/src/main/resources/scripts/docker-setup.sh [hunk 10]
```diff
@@ -0,0 +1,138 @@
+#!/bin/bash
```
Comment (Michael-Baylis) [inline galasa-extensions-parent/dev.galasa.ecosystem.builder/src/main/resources/scripts/docker-setup.sh]: Transient files should not be committed

File: galasa-extensions-parent/dev.galasa.ecosystem.builder/src/main/resources/scripts/docker-setup.sh [hunk 11]
```diff
@@ -0,0 +1,138 @@
...
+
+source docker/config.properties
+
+echo "Pulling etcd"
+docker pull quay.io/coreos/etcd:v3.2.25
```
Comment (Michael-Baylis) [inline galasa-extensions-parent/dev.galasa.ecosystem.builder/src/main/resources/scripts/docker-setup.sh]: dont need to pull.   the run will sort that out.   Also,  dont want to pull incase they have used a future "load" command to load the images

File: galasa-extensions-parent/dev.galasa.ecosystem.builder/src/main/resources/scripts/docker-update.sh [hunk 12]
```diff
@@ -0,0 +1,105 @@
+#!/bin/bash
```
Comment (Michael-Baylis) [inline galasa-extensions-parent/dev.galasa.ecosystem.builder/src/main/resources/scripts/docker-update.sh]: Transient files should not be committed

**Comment by Roavill2** [comment]
Changes made to code to meet requirements from first attempt

**Comment by Roavill2** [comment]
Closing as opening new PR.

**Pull request closed without merging**